irr/include/irrString.h

```diff
@@ -165,21 +166,25 @@ class string
 	}
 
 	//! Assignment operator for strings, ASCII and Unicode
-	template <class B>
-	string<T> &operator=(const B *const c)
+	template <class B, std::enable_if_t<std::is_array_v<B[]>, bool> = true>
+	string<T> &operator=(const B * c)
 	{
 		if (!c) {
 			clear();
 			return *this;
 		}
-
-		// no longer allowed!
-		_IRR_DEBUG_BREAK_IF((void *)c == (void *)c_str());
-
-		u32 len = calclen(c);
-		str.resize(len);
-		for (u32 l = 0; l < len; ++l)
-			str[l] = (T)c[l];
+		
+        _IRR_DEBUG_BREAK_IF(
```

Comment:
Where did sfan's `if constexpr` go? That's necessary to only run this check in case sizes differ. Otherwise this is well defined and we want to allow it.

Reply:
It wouldn't compile for me

Reply:
Could you be more specific? I'd be surprised if `if constexpr (sizeof(T) != sizeof(B)) { ... }` didn't compile. Comparing two sizes at comptime should definitely be possible. Does it compile if you remove the `constexpr`? It isn't really needed because the compiler should definitely be able to constant fold that away anyways.

Reply:
i think it was the casting thing that wasn't compiling. I can just use my casting code but bring back his constexpr check

Reply:
@appgurueu In theory, by checking if B is a pointer (which is.. kind of already is? But the template check ensures the user can't just pass in some bullshit (not the weird one you see)), the size should pretty much always be the same here. It's completely unnecessary. I see no purpose in this.

Again, why are we so focused on pleasing Bjarne on a function we should kill out later? It feels like we are wasting time on semantics for code that we should just ship already.

Reply:
> In theory, by checking if B is a pointer

The function signature uses `const B *const` so of course `B` itself is (usually) not a pointer but rather the "character" type of the pointer we're copying from.

> Again, why are we so focused on pleasing Bjarne on a function we should kill out later?

Personally I don't feel strongly about this sanity check (it's probably unnecessary), but it doesn't cost much so why not.

> I'm tired of rebuilding Minetest just to get this to work :')

Don't worry this is close.